sway-core/src/lib.rs

```diff
@@ -205,10 +205,9 @@ pub enum BytecodeCompilationResult {
     },
 }
 
-pub fn compile_to_ast(
-    input: Arc<str>,
+pub fn parsed_to_ast(
+    parsed: CompileResult<ParseProgram>,
```

Comment:
This should probably be:

```suggestion
    parsed: ParseProgram,
```

To make this change, we'll also need to change the `compile_to_ast` body to:

```rust
    parse(input, build_config).flat_map(|parsed| {
        parsed_to_ast(parsed, initial_namespace)
    })
```

---

The `CompileResult`'s `flat_map` method should probably be called `and_then` to better align with the `std` `Option` and `Result` methods, but that's a future PR.

Reply:
But don't we still want to carry any errors and warnings across from the parsing stage?

Reply:
The result returned from `sway_core::parse` should have already been handled by the user as they likely require access to the `ParseProgram` within the result (otherwise they would have just called `compile_to_ast` in the first place).

Reply:
Thanks @mitchmindtree I've changed `parsed_to_ast` to take a `ParseProgram` instead of a `CompileResult<ParseProgram>`.

I've also tried to retain the original behaviour of carrying forward any warnings and errors found during the parsing stage when calling `compile_to_ast`.